### PR TITLE
Removed Entra App Secrets from Storage tests

### PR DESCRIPTION
**Pull request by seanmcc-msft**
This PR removes the .NET Storage SDK's test dependency on Entra ID app secrets, and removed the requirement that test accounts allow public access.

Note that once this PR merges, the Entra ID-related items in TestConfigurations.xml need to be deleted.

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
@schaabss please let me know when the .NET Storage live test pipeline is ready to accept this PR.

File: sdk/storage/tests.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ extends:
       Preview:
         SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources-preview)
         ServiceConnection: azure-sdk-tests
+        SubscriptionConfigurationFiles:
+          - sdk/storage/sub-config-azure-preview.json
```
Comment (danieljurek) [inline sdk/storage/tests.yml:16]: This path does not match the path of the file created: sdk/storage/storage-preview-public-msft.json

File: sdk/storage/tests.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ extends:
       Preview:
         SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources-preview)
         ServiceConnection: azure-sdk-tests
+        SubscriptionConfigurationFiles:
```
Comment (weshaggard) [inline sdk/storage/tests.yml:15]:
```suggestion
        SubscriptionConfigurationFilePaths:
```

**Comment by seanmcc-msft** [comment]
TODO try making Managed Disk and Immutability with Versioning tests live again.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs [hunk 3]
```diff
@@ -254,6 +254,7 @@ public async Task Ctor_DefaultAudience()
         }
 
         [RecordedTest]
+        [PlaybackOnly("DefaultAzureCredential does not play nicely with this test.")]
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs:257]: Should we create a github issue to eventually reenable this test?

File: sdk/storage/Azure.Storage.Queues/tests/QueueTestBase.cs [hunk 4]
```diff
@@ -12,6 +12,7 @@
 using Azure.Storage.Sas;
 using Azure.Storage.Test;
 using Azure.Storage.Test.Shared;
+using Microsoft.Extensions.Options;
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Queues/tests/QueueTestBase.cs:15]: NIT - looks like visual studio added this by accident

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml [hunk 5]
```diff
@@ -13,181 +13,168 @@
 
   <TenantConfigurations>
     <!-- A storage account with an Encryption Scope, Blob Versioning, and Last Access Time enabled -->
-    <!-- Replace [ACCOUNT] and [ACCOUNT-KEY] -->
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml:16]: NIT why are the brackets removed?

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml [hunk 6]
```diff
@@ -13,181 +13,168 @@
...
       <Name>DefaultManagedDisk</Name>
-      <DiskNamePrefix>[DISK NAME PREFIX]</DiskNamePrefix>
-      <ResourceGroupName>[RESOURECE GROUP NAME]</ResourceGroupName>
-      <SubsriptionId>[SubscriptionId]</SubsriptionId>
-      <Location>[Location]</Location>
-      <ActiveDirectoryApplicationId>[ActiveDirectoryApplicationId]</ActiveDirectoryApplicationId>
-      <ActiveDirectoryApplicationSecret>[ActiveDirectoryApplicationSecret]</ActiveDirectoryApplicationSecret>
-      <ActiveDirectoryTenantId>[ActiveDirectoryTenantId]</ActiveDirectoryTenantId>
+      <DiskNamePrefix>DISK NAME PREFIX</DiskNamePrefix>
+      <ResourceGroupName>RESOURECE GROUP NAME</ResourceGroupName>
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml:173]: NIT change  RESOURECE  to RESOURCE

**Review by amnguye — APPROVED** [review]
looks good - Had some NIT comments

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml (hunk 5 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml:16]: My understanding is that `[` and `]` are invalid XML characters.  This means that before this change, you had fill in all the tests accounts, even if you only needed to run one type of test.

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml (hunk 6 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml]: Done

File: sdk/storage/Azure.Storage.Queues/tests/QueueTestBase.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Queues/tests/QueueTestBase.cs:15]: Fixed.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs]: https://github.com/Azure/azure-sdk-for-net/issues/44967, and updated the `PlaybackOnly` annotations to refer to this issue.

**Pull request merged**